### PR TITLE
Fix Mono Pack and restore test

### PR DESCRIPTION
**Pull request by zhili1208**
Path fix

### CODE REVIEW

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs [hunk 1]
```diff
@@ -26,11 +26,11 @@ public void PackCommand_IncludeExcludePackageFromNuspec()
             {
                 // Arrange
                 Util.CreateFile(
-                    Path.Combine(workingDirectory, "contentFiles/any/any"),
+                    Path.Combine(workingDirectory, "contentFiles","any","any"),
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: nit: extra space after `,`

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs [hunk 2]
```diff
@@ -418,10 +418,10 @@ public void PackCommand_PackRuntimesRefNativeNoWarnings()
                     files,
                     new string[]
                     {
-                            @"lib\uap10.0\a.dll",
-                            @"native\a.dll",
-                            @"ref\uap10.0\a.dll",
-                            @"runtimes\win-x86\lib\uap10.0\a.dll",
+                            Path.Combine("lib","uap10.0","a.dll"),
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: nit: space after `,`

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs [hunk 3]
```diff
@@ -1692,11 +1692,16 @@ public void PackCommand_OutputDirectorySuffixes(string suffix)
             {
                 // Arrange
                 CreateTestProject(workingDirectory, "proj1", new string[] { });
+                if(!RuntimeEnvironmentHelper.IsWindows && RuntimeEnvironmentHelper.IsMono)
+                {
+                    suffix = suffix.Replace('\\', Path.DirectorySeparatorChar);
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: this replacement is always safe, not matter what platform.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs [hunk 4]
```diff
@@ -1692,11 +1692,16 @@ public void PackCommand_OutputDirectorySuffixes(string suffix)
             {
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:1694]: Can we add a test case for a single slash?

Comment (joelverhagen) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:1694]: Also, what happens when we pass the "wrong" slash? Like a backslash `\` on Linux/Mac or a forward slash `/` on Windows?

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs [hunk 5]
```diff
@@ -533,7 +534,7 @@ public void RestoreCommand_FromPackagesConfigFileWithOptionSolutionDir()
             // Arrange
             var nugetexe = Util.GetNuGetExePath();
 
-            using (var workingPath = TestFileSystemUtility.CreateRandomTestFolder())
+            using(var workingPath = TestFileSystemUtility.CreateRandomTestFolder())
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs]: nit: no need to remove that space

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs [hunk 6]
```diff
@@ -1785,12 +1789,11 @@ var nugetFolderAtSolutionDirectory
 
                 // Assert
                 Assert.Equal(0, r.Item1);
-
                 var dllPath = Path.Combine(workingPath, "outputDir", "packageA.1.1.0", "lib", "net45", "A.dll");
                 var dllFileInfo = new FileInfo(dllPath);
                 Assert.True(File.Exists(dllFileInfo.FullName));
                 Assert.Equal(entryModifiedTime, dllFileInfo.LastWriteTime);
-            }
+            }
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs]: nit: no need for the extra trailing whitespace here

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs [hunk 7]
```diff
@@ -1938,10 +1942,11 @@ public void RestoreCommand_SourceLoggingFileSource()
 
                 // Assert
                 Assert.Equal(0, r.Item1);
-                var target = new PackagePathResolver(Path.Combine(workingPath, @"outputDir"), false);
+                var target = new PackagePathResolver(Path.Combine(workingPath, @"outputDir"), true);
                 var packageFilePath = target.GetInstalledPackageFilePath(identity);
-                Assert.True(File.Exists(packageFilePath));
+                var test = PackagePathHelper.GetPackageLookupPaths(identity, target);
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs]: is the `test` local used?

**Review by joelverhagen — CHANGES_REQUESTED** [review]

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs (hunk 7 above)
Comment (zhili1208) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs]: good catch

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs [hunk 8]
```diff
@@ -778,9 +778,9 @@ public class Class1
                     files,
                     new string[]
                     {
-                        @"content\proj1_file2.txt",
-                        @"lib\net40\proj1.dll",
-                        @"lib\net40\proj2.dll"
+                        Path.Combine("content","proj1_file2.txt"),
+                        Path.Combine("lib","net40","proj1.dll"),
+                        Path.Combine("lib","net40","proj2.dll")
```
Comment (mishra14) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: nit: Format the doc to fix all the `, `?

Comment (mishra14) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: nit: Format the doc to fix the `,` all across.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs [hunk 9]
```diff
@@ -1716,7 +1719,7 @@ public void PackCommand_OutputDirectorySuffixes(string suffix)
                     files,
                     new string[]
                     {
-                        @"lib\net40\proj1.dll"
+                        Path.Combine("lib","net40","proj1.dll")
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: nit: more missing spaces

**Comment by mishra14** [comment]
Please address @joelverhagen 's comments! 

otherwise LGTM!

**Pull request closed without merging**